workspaces/ui/src/components/diff/v2/AddUrlModal.js

```diff
@@ -89,40 +89,45 @@ export const NewUrlModal = withNavigationContext(withRfcContext(({pushRelative,
     handleCommands(...commands);
 
     //redirect
-    const to = `/diff/${captureId}/paths/${lastParentPathId}/methods/${newUrl.method}`
-    pushRelative(to)
+    const to = `${captureId}/paths/${lastParentPathId}/methods/${newUrl.method}`
+    history.push(to)
   };
 
   const regex = completePathMatcherRegex(pathStringToPathComponents(pathExpression));
   const matches = regex.exec(newUrl.path) !== null;
   const matchingUrls = Array.from(allUnmatchedPaths.filter((url) => regex.exec(url) && url !== newUrl.path));
 
-  const namingDialog = (
-    <Dialog open={open} onClose={handleClose} fullWidth maxWidth="md" aria-labelledby="form-dialog-title">
-      <DialogTitle>Add New Endpoint</DialogTitle>
-      <DialogContent style={{marginTop: -20}}>
-        <PathAndMethod method={newUrl.method} path={pathExpression}/>
-        <DialogContentText style={{marginTop: 12}}>
-          What does this endpoint do?
-        </DialogContentText>
-        <TextField value={purpose} onChange={(e) => setPurpose(e.target.value)} autoFocus fullWidth/>
-      </DialogContent>
-      <DialogActions>
-        <Button onClick={() => setNaming(false)}>
-          Back
-        </Button>
-        <Button onClick={handleCreate} color="secondary" disabled={!matches}
-                endIcon={<NavigateNextIcon/>}>Finish</Button>
-      </DialogActions>
-    </Dialog>
-  );
+  function NamingDialog() {
+    const [purpose, setPurpose] = React.useState('');
```

Comment:
@acunniffe I moved this into here, since it was re-rendering the parent on every change before. Let's try to stay conscious of that